.travis.yml

```diff
@@ -2,25 +2,38 @@ language: csharp
 solution: JSIL.sln
 
 before_install:
-  - wget -P /tmp/avim -N https://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/latest-mozilla-central/jsshell-linux-x86_64.zip
-  - unzip -d /tmp/js /tmp/avim/jsshell-linux-x86_64.zip
+  - wget -P /tmp/ -N https://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/latest-mozilla-central/jsshell-linux-x86_64.zip
+  - unzip -d /tmp/js /tmp/jsshell-linux-x86_64.zip
   - export PATH=$PATH:/tmp/js
 
 install:
   - export NODEJS=$(dirname `nvm which $NODEJS_VERSION`)
   - export FSHARPINSTALLDIR=$PWD/packages/FSharp.Compiler.Tools.4.0.0.1/tools/
-  - ci/download-asset.sh packages .
   - nuget restore JSIL.sln
   - nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory packages 
   - nuget install FSharp.Compiler.Tools -Version 4.0.0.1 -OutputDirectory packages 
-  - ci/upload-asset.sh packages packages &
+
+before_script:
+  - mv /tmp/JSIL_Tests "/tmp/JSIL Tests" || true
```

Comment:
I thought about it, but it don't look as critical issue for me. All precompiled tests directories will be merged (maybe not on first run, but not later then 3rd run - you even can restart each sub-build sequentially). After it cache will store precompiled test cases for all tests.